src/x11/i3/ipc/listener.cpp

```diff
@@ -0,0 +1,43 @@
+#include "listener.hpp"
+#include <utility>
+
+#include <qbytearrayview.h>
+#include <qcontainerfwd.h>
+#include <qdatastream.h>
+#include <qobject.h>
+#include <qsysinfo.h>
+#include <qtenvironmentvariables.h>
+#include <qtmetamacros.h>
+
+#include "connection.hpp"
+
+namespace qs::i3::ipc {
+
+I3IpcListener::~I3IpcListener() { delete this->i3Ipc; }
+
+void I3IpcListener::onPostReload() { this->startListening(); }
+
+QList<QString> I3IpcListener::subscriptions() const { return this->mSubscriptions; }
+void I3IpcListener::setSubscriptions(QList<QString> subscriptions) {
+	if (this->mSubscriptions == subscriptions) return;
+	this->mSubscriptions = std::move(subscriptions);
+
+	emit this->subscriptionsChanged();
+	this->startListening();
+}
+
+void I3IpcListener::startListening() {
+	if (this->i3Ipc != nullptr || this->mSubscriptions.isEmpty()) return;
```

Comment:
This wont allow subscriptions to be changed during operation. Why not replace the I3Ipc object instead?